### ISSUE TITLE
Worker Thread  still alive even it end its works 

### ISSUE THREAD

**Issue by aharondavid** [emscripten-core/emscripten#16761]
Worker Thread that created by pthread_create, pthread_detach, seem in the debugger (Chrome Dev Tool) although it ended and returned.
it removed only when calling to pthread_kill

![image](https://user-images.githubusercontent.com/41375082/164223790-d7b82e61-eab7-425f-85d0-2151b56d2f47.png)

**Version of emscripten/emsdk:**
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.4 (14cd48e6ead13b02a79f47df1a252abc501a3269)
clang version 15.0.0 (https://github.com/llvm/llvm-project ce5588fdf478b6af724977c11a405685cebc3d26)
Target: wasm32-unknown-emscripten
Thread model: posix

**Comment by aharondavid** [comment]
probably the same as : https://github.com/emscripten-core/emscripten/issues/16763

**Comment by sbc100** [comment]
emscripten doesn't currently ever destroy workers.. it keeps them around for future pthreads to run on.

**Comment by aharondavid** [comment]
Hi,
Good to know! I think that there is no any treatment to that important fact in any emscripten documentation.

**Comment by kripken** [comment]
I agree it would be good to document this. Let's reopen this issue for that.

**Comment by zekiparlak** [comment]
So there is no meaning using pthread_kill ?

**Comment by sbc100** [comment]
Emscripten uses a worker pool regardless of whether you specify `PTHREAD_POOL_SIZE`.  The purpose of `PTHREAD_POOL_SIZE` is simply to pre-populate the pool.

If you create 10 threads and then destroy them you will end up with a pool of 10 workers.   Those workers are kept around a long as your app is running.
